docs/detections/alert-suppression.asciidoc

```diff
@@ -23,37 +25,44 @@ NOTE: Alert suppression is not available for Elastic prebuilt rules. However, if
+* New terms rule: In *Suppress alerts by*, enter 1-3 field names to group events by the fields' values.
+
 --
 +
 [NOTE]
 ======
 Fields with multiple values are handled as follows:
 
 * **Custom query or threshold rules** - If you specify a field with multiple values, an alert grouping is created for each value. For example, if you suppress alerts by `destination.ip` of `[127.0.0.1, 127.0.0.2, 127.0.0.3]`, alerts will be suppressed separately for each value of `127.0.0.1`, `127.0.0.2`, and `127.0.0.3`. 
-* **Indicator match rule** - If you specify a field with multiple values, an alert grouping is created for alerts that contain the array you specified. For example, if you suppress alerts by `destination.ip` of `[127.0.0.1, 127.0.0.2, 127.0.0.3]`, alerts that contain this array are grouped and suppressed.
+* **Indicator match rule, event correlation (non-sequence queries only), or new terms rule** - If you specify a field with multiple values, an alert grouping is created for alerts that contain the array you specified. For example, if you suppress alerts by `destination.ip` of `[127.0.0.1, 127.0.0.2, 127.0.0.3]`, alerts that contain this array are grouped and suppressed.
 
 ======
 
 . If available, select how often to create alerts for duplicate events:
 +
+NOTE: Both options are available for custom query, indicator match, event correlation, and new terms rules. Threshold rules only have the *Per time period* option.
++
 --
-* *Per rule execution*: (Only available for custom query rules and indicator match rules) Create an alert each time the rule runs and meets its criteria.
-* *Per time period*: Create one alert for all qualifying events within a specified time window, beginning when the rule first meets its criteria and creates the alert. (This is the only option available for threshold rules.)
+* *Per rule execution*: Create an alert each time the rule runs and meets its criteria.
+* *Per time period*: Create one alert for all qualifying events within a specified time window, beginning when the rule first meets its criteria and creates the alert.
 +
 For example, if a rule runs every 5 minutes but you don't need alerts that frequently, you can set the suppression time period to a longer time, such as 1 hour. If the rule meets its criteria, it creates an alert at that time, and for the next hour, it'll suppress any subsequent qualifying events.
 +
 image::images/alert-suppression-options.png[Alert suppression options,400]
 --
 
-. (Only available for custom query rules and indicator match rules) Under *If a suppression field is missing*, choose how to handle events with missing suppression fields (events in which one or more of the *Suppress alerts by* fields don't exist):
+. Under *If a suppression field is missing*, choose how to handle events with missing suppression fields (events in which one or more of the *Suppress alerts by* fields don't exist):
```

Comment:
I don't know if this is helpful for docs, but when I was trying to understand this feature, I found it useful to compare it to the [missing](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-terms-aggregation.html#_missing_value_5) option on a terms aggregation:

* The default suppression behavior is effectively `missing: 'other'`: all alerts missing the suppression field will be suppressed into a single "other" bucket
* The alternate suppression behavior is effectively to omit `missing`, causing alerts without the suppression field to be ignored (from the perspective of suppression, i.e. not suppressed).

cc @vitaliidm

Reply:
> The default suppression behavior is effectively missing: 'other': all alerts missing the suppression field will be suppressed into a single "other" bucket

Yes, value would be `null` in this case

> The alternate suppression behavior is effectively to omit missing, causing alerts without the suppression field to be ignored (from the perspective of suppression, i.e. not suppressed).

yes, they will not be suppressed, but still created

Reply:
@rylnd - I didn't know there was comparable Elasticsearch functionality, that's cool. Do you think we need to improve the current descriptions for the options under the `Suppress and group alerts for events with missing fields` section? If so, do you have any revisions in mind or suppression-related examples that can illustrate how the options work?

@vitaliidm - in the docs for the `Suppress and group alerts for events with missing fields` option, we write that "missing fields get a null value, which is used to group and suppress alerts." I don't fully understand this. Are suppression fields with `null` values added to event documents before alerts are created? Or are event documents treated as though they have the suppression fields (i.e., the event document isn't modified at all), and then grouped and suppressed during alert creation?

Reply:
> Or are event documents treated as though they have the suppression fields (i.e., the event document isn't modified at all), and then grouped and suppressed during alert creation?

they treated as they have suppression field with value `null`

Reply:
Chatted with @rylnd about his [comment](https://github.com/elastic/security-docs/pull/5057#discussion_r1561826946), and we'll be leaving the docs as they are for now. If we were to expand them in the future, it'd be helpful to add some concrete examples of both options. 

Also chatted with @vitaliidm about how events that are missing suppression fields are treated. 

If an event is missing a suppression field, that information is tracked in the `kibana.alert.suppression.terms.field` alert field and denoted by the field value `null`. For example, if a rule is configured to suppress by the `kibana.alert.rule.name` and `host.name` fields, and the user also selected the option to **Suppress and group alerts for events with missing fields**, the document for the suppression alert will contain the following values within the `kibana.alert.suppression.terms.field` field:

- Unique values for events that contain the `kibana.alert.rule.name` field
- Unique values for events that contain the `host.name` field
- The `null` value for events that don't contain the `kibana.alert.rule.name` or the `host.name` fields

Example of what it might look like in the alert doc:
<table>
<tr> 
 <td> Field
 <td> Value
<tr> 
 <td>kibana.alert.suppression.terms.field
 <td> kibana.alert.rule.name 

host.name
null
</table>

Reply:
Oops, that was a field name typo. Regardless, thanks for the additional examples and explanation.